neware_parser/PlotEngine.py

```diff
@@ -0,0 +1,120 @@
+import pickle
+
+import matplotlib.patches as mpatches
+
+# TODO(harvey) duplicate in plot.py
+COLORS = [
+    (.4, .4, .4),
+
+    (1., 0., 0.),
+    (0., 0., 1.),
+    (0., 1., 0.),
+
+    (.6, 0., .6),
+    (0., .6, .6),
+    (.6, .6, 0.),
+
+    (1., 0., .5),
+    (.5, 0., 1.),
+    (0., 1., .5),
+    (0., .5, 1.),
+    (1., .5, 0.),
+    (.5, 1., 0.),
+]
+
+
+class PlotEngine:
+
+    @staticmethod
+    def pickle_load_scale(filename: str) -> tuple:
+        f = open(filename, "rb")
+        patches = pickle.load(f)
+        keys = pickle.load(f)
+        scales = pickle.load(f)
+        cycles = pickle.load(f)
+        f.close()
+
+        return patches, keys, scales, cycles
+
+    @staticmethod
+    def plot_scale(filename: str, fig, offset: int) -> None:
+        """ Plot scale from the given pickle
+
+        Args:
+            filename (str): Filename (including path) to the pickle file
+            fig: The figure on which to plot scale
+            offset (int): The offset on the figure for the scale plot
+        """
+
+        patches, keys, scales, cycles = PlotEngine.pickle_load_scale(filename)
+        ax1 = fig.add_subplot(6, 1, 1 + offset)
```

Comment:
this offset business is super finicky.
instead, pass in ax1.
try to make the pieces reusable. This plot_scale thing does a single panel plot. if you pass in ax1 (the axis), someone else is in charge of the layout. plot_scale can be called from a bunch of layouts.